Build against different mbed-os revisions & hw smoke

kimlep01 opened the pull request:
Build against different mbed-os revisions & hw smoke

* Added ability to build against different mbed-os branches (default being master)
- Triggered from Jenkins "Build with parameters"
* Removed non compatible Ublox & mcr20 build combination
* Added a simple hw smoke test as the final stage with K64F & atmel combination
- Smoke test is not 100% reliable yet (ONME-2882).

@SeppoTakalo @artokin

SeppoTakalo commented:
From: https://jenkins-internal.mbed.com/view/Morpheus_CI/job/ARMmbed/job/mbed-os-example-mesh-minimal/view/Pull%20Requests/job/PR-32/2/consoleText

| CFG | status |
|-----|--------|
|K64F_GCC_ARM_ATMEL_6lp | OK |
|K64F_ARM_ATMEL_6lp | fail |
|K64F_IAR_ATMEL_6lp | OK |

Check the status. I think we have spotted a real error.

From the previous run: https://jenkins-internal.mbed.com/view/Morpheus_CI/job/ARMmbed/job/mbed-os-example-mesh-minimal/view/Pull%20Requests/job/PR-32/1/consoleText

Same results.
Its the ARM build that fails.

kimlep01 commented:
Status: ready
* Added boolean parameter to run smoke tests (default disabled)
* Added Thread test suites
* Smoke test output: smoke_RaasPort_suiteName

Results: https://jenkins-internal.mbed.com/job/ARMmbed/job/mbed-os-example-mesh-minimal/view/Pull%20Requests/job/PR-32/
# Build 13: With tests
# Build 15: without tests (14 had issues with IAR slave going offline)
